### PR TITLE
Homepage!

### PR DESCRIPTION
**Pull request by dfrasier**
Includes…
* Router updates (url for the page)
* Top section with hero image and buttons.
* Budget Functions section with large image of the treasury as the background.
* Responsive TreeMap.
* “More to explore section” with placeholder links to other features on the site.

### CODE REVIEW

File: src/js/components/homepage/LinksSection.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                <h4
+                    className="links-section-primary-title">
+                    Check out these sections for more views into our data.
+                </h4>
+                <div className="links-section-box-wrap">
+                    <LinksSectionBox
+                        icon="recipient"
+                        text="Get insights into specific recipients."
+                        linkText="Find a recipient"
+                        linkUrl="http://google.com" />
```
Comment (mab6bf) [inline src/js/components/homepage/LinksSection.jsx]: Let's make these URLs empty strings for now

File: src/js/components/homepage/LinksSectionBox.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    linkText: React.PropTypes.string,
+    linkUrl: React.PropTypes.string
+};
+
+export default class LinksSectionBox extends React.Component {
+
+    render() {
+        let icon = this.props.icon;
+
+        switch (icon) {
```
Comment (mab6bf) [inline src/js/components/homepage/LinksSectionBox.jsx]: For the purposes of keeping `render()` small, let's move this `switch` statement into it's own function, above `render()` (something like `generateIcon()`).

File: src/js/components/homepage/visualizations/TreeMap.jsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        let tileStyle = d3.treemapBinary;
+        if (this.state.windowWidth < 768) {
+            tileStyle = d3.treemapSlice;
+        }
+        const treemap = d3.treemap()
+            .round(true)
+            .tile(tileStyle)
+            .size([this.state.visualizationWidth, 565])(root).leaves();
+
+        const colors = [
```
Comment (mab6bf) [inline src/js/components/homepage/visualizations/TreeMap.jsx]: We don't need to recreate this array with every `render()` cycle - let's move this near `propTypes` as it's own structure.

File: src/js/components/homepage/visualizations/TreeMap.jsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+            // width changed, update the visualization width
+            this.setState({
+                windowWidth,
+                visualizationWidth: this.sectionWrapper.offsetWidth
+            });
+        }
+    }
+
+    render() {
+        // put the data through d3's hierarchy system to sum and sort it
```
Comment (mab6bf) [inline src/js/components/homepage/visualizations/TreeMap.jsx]: Let's move the generation of `finalNodes` out of `render()` and into it's own method above `render()`.

File: src/js/components/search/table/ResultsTable.jsx [hunk 5]
```diff
@@ -145,6 +145,7 @@ export default class ResultsTable extends React.PureComponent {
     }
 
     render() {
+        console.log(this.props);
```
Comment (mab6bf) [inline src/js/components/search/table/ResultsTable.jsx]: Remove this, please

File: src/js/containers/router/RouterRoutes.jsx [hunk 6]
```diff
@@ -35,12 +34,16 @@ const goToPage = (location, replace) => {
...
-    // },
     indexRoute: {
-        onEnter: (nextState, replace) => replace('/search')
+        getComponent(nextState, cb) {
+            require.ensure([], (require) => {
+                cb(null, require('../homepage/HomepageContainer').default);
+            });
+        }
     },
+    // indexRoute: {
```
Comment (mab6bf) [inline src/js/containers/router/RouterRoutes.jsx]: Do we still need this code in the future, or can we get rid of it?

File: src/js/containers/homepage/HomepageContainer.jsx [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+import TreeMapIntro from 'components/homepage/TreeMapIntro';
+import LinksSection from 'components/homepage/LinksSection';
+import SearchSection from 'components/homepage/SearchSection';
+import Header from '../../components/sharedComponents/header/Header';
+import Footer from '../../components/sharedComponents/Footer';
+
+export default class HomepageContainer extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
```
Comment (mab6bf) [inline src/js/containers/homepage/HomepageContainer.jsx]: Will this data eventually be populated by an API call? Is that why this file is a container and not a component?

Comment (ejgullo) [inline src/js/containers/homepage/HomepageContainer.jsx]: negatory, static data.

File: src/js/components/homepage/LinksSection.jsx (hunk 1 above)
Comment (ejgullo) [inline src/js/components/homepage/LinksSection.jsx]: 👍

File: src/js/components/homepage/LinksSectionBox.jsx (hunk 2 above)
Comment (ejgullo) [inline src/js/components/homepage/LinksSectionBox.jsx]: 👍

File: src/js/components/homepage/visualizations/TreeMap.jsx (hunk 3 above)
Comment (ejgullo) [inline src/js/components/homepage/visualizations/TreeMap.jsx]: 👍

File: src/js/components/search/table/ResultsTable.jsx (hunk 5 above)
Comment (ejgullo) [inline src/js/components/search/table/ResultsTable.jsx]: 👍

File: src/js/containers/router/RouterRoutes.jsx (hunk 6 above)
Comment (ejgullo) [inline src/js/containers/router/RouterRoutes.jsx]: nixing it, forgot to remove :(

File: src/js/containers/homepage/HomepageContainer.jsx (hunk 7 above)
Comment (ejgullo) [inline src/js/containers/homepage/HomepageContainer.jsx]: moving to component 👍

File: src/js/components/homepage/visualizations/TreeMap.jsx (hunk 4 above)
Comment (ejgullo) [inline src/js/components/homepage/visualizations/TreeMap.jsx]: 👍

File: src/js/components/homepage/TreeMapIntro.jsx [hunk 8]
```diff
@@ -0,0 +1,20 @@
+/**
+ * TreeMap.jsx
```
Comment (ejgullo) [inline src/js/components/homepage/TreeMapIntro.jsx]: 👍

**Pull request merged**